proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -981,6 +971,18 @@ public void spoofChatInput(String input) {
     }
   }
 
+  /**
+   * Get the ResourcePackHandler corresponding to the player's version.
+   *
+   * @return the ResourcePackHandler of this player
+   */
+  public ResourcePackHandler resourcePackHandler() {
```

Comment:
What is the risk that this is called across multiple threads? Do we need to guard this?

Reply:
I doubt that in this case anything will happen, but of course, I will make the variable volatile.
I did the implementation this way to avoid creating the respective implementation if the server does not use resource packs

Reply:
volatile won't fix the concern if we do expect it to be called from various threads, which if it was just purely passthrough wouldn't be much of a deal as it would happen purely on the event loop, if a plugin calls it, low risk, but there is a chance we could double init